### PR TITLE
Cli cwd changes

### PR DESCRIPTION
**Pull request by deontologician**
Implements #376 

```
 checkProjectName
    when passed null for a directory
      ✓ doesn't change directory
      ✓ throws if the cwd has invalid chars
      ✓ sets projectName to dehyphenated cwd if fixable
    when passed "." as a directory
      ✓ doesn't change directory
      ✓ throws if the cwd has invalid chars
      ✓ sets projectName to dehyphenated cwd if fixable
    when passed a non-existing directory
      ✓ creates the directory when name is valid
      ✓ creates the directory when name is fixable
      ✓ throws an error if the name is not fixable
    when passed an existing directory
      ✓ errors if given an invalid projectName
      ✓ changes directory  if the name is good
      ✓ changes directory if the name is fixable
```

---

This change is [Reviewable](https://reviewable.io/reviews/rethinkdb/horizon/472)

### CODE REVIEW

**Comment by marshall007** [comment]
One minor thing on this, I think you should be able to specify an "invalid" directory name (containing hyphens) and we should only worry about sanitizing the project name in the `config.toml`.

**Comment by deontologician** [comment]
Good point

**Comment by deontologician** [comment]
Ok, this is ready to review. Any takers @Tryneus or @dalanmiller ?

File: cli/src/main.js [hunk 1]
```diff
@@ -72,36 +72,55 @@ const done_cb = (options) => (err) => {
   }
 };
 
+function runCommand(command, options, done) {
```
Comment (Tryneus) [inline cli/src/main.js]: since `done` is always `done_cb(options)`, that parameter is extraneous.

**Comment by deontologician** [comment]
I went ahead and made the refactors to cli main.js, since I was feeling less lazy today

File: cli/src/main.js [hunk 2]
```diff
@@ -14,96 +14,59 @@ const getSchemaCommand = require('./get-schema');
...
-  help: 'Get the schema from a horizon database',
-});
-
-const setSchemaParser = subparsers.addParser('set-schema', {
-  addHelp: true,
-  help: 'Set the schema in a horizon database',
-});
+// Mapping from command line strings to modules. To add a new command,
+// add an entry in this object, and create a module with the following
+// exported:
```
Comment (dalanmiller) [inline cli/src/main.js]: "following functions exported"?

**Comment by dalanmiller** [comment]
:lgtm:

**Pull request merged**